### PR TITLE
Add possibility to reset column types when changing the ordinal threshold

### LINKED ISSUE #2426: [Bug]: Change threshhold default for nominal/ordinal OR allow summing up ordinal (Likert) data
### JASP Version

0.18.1

### Commit ID

_No response_

### JASP Module

Unrelated

### What analysis are you seeing the problem on?

data import / data wrangling

### What OS are you seeing the problem on?

Other

### Bug Description

image

The threshold is currently set to 10. 
But if I import a typical social sciences data set with many Likert scaled items, it would be much more user friendly, if the default threshold would be set to 4 (below the value of 5, which is the default for Likert scaled items).

Why?
Usually you will need to sum up many items with each other. But you cannot compute a sum with Likert scaled data, because it is always imported as ordinal. You have then to change the scale of dozens of items/variables from ordinal to scale - which is very inconvenient.
Another solution would be just to allow summing up ordinal data.

### Expected Behaviour

Summing up Likert scaled items should be easy.

### PR DESCRIPTION
**Pull request by boutinb**
Fixes https://github.com/jasp-stats/jasp-issues/issues/2426

If you change the threshold between Categorical and Scale, it might be handy to reset the variable types with this new value.

I have also redesigned a bit the Preferences/Data to have the same look as the other preferences menus.

### CODE REVIEW

**Comment by shun2wang** [comment]
Seems the build test failed.

**Comment by JorisGoosen** [comment]
```
[ 70%] Building CXX object Desktop/CMakeFiles/JASP.dir/gui/preferencesmodel.cpp.o
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:81:62: error: no declaration matches ‘bool PreferencesModel::customThresholdScale() const’
   81 | #define GET_PREF_FUNC(TYPE, NAME, SETTING, TO_FUNC)     TYPE PreferencesModel::NAME() const { return Settings::value(SETTING).TO_FUNC; }
      |                                                              ^~~~~~~~~~~~~~~~
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:81:62: note: in definition of macro ‘GET_PREF_FUNC’
   81 | #define GET_PREF_FUNC(TYPE, NAME, SETTING, TO_FUNC)     TYPE PreferencesModel::NAME() const { return Settings::value(SETTING).TO_FUNC; }
      |                                                              ^~~~~~~~~~~~~~~~
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:101:1: note: in expansion of macro ‘GET_PREF_FUNC_BOOL’
  101 | GET_PREF_FUNC_BOOL(     customThresholdScale,           Settings::USE_CUSTOM_THRESHOLD_SCALE                            )
      | ^~~~~~~~~~~~~~~~~~
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:81:62: note: no functions named ‘bool PreferencesModel::customThresholdScale() const’
   81 | #define GET_PREF_FUNC(TYPE, NAME, SETTING, TO_FUNC)     TYPE PreferencesModel::NAME() const { return Settings::value(SETTING).TO_FUNC; }
      |                                                              ^~~~~~~~~~~~~~~~
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:81:62: note: in definition of macro ‘GET_PREF_FUNC’
   81 | #define GET_PREF_FUNC(TYPE, NAME, SETTING, TO_FUNC)     TYPE PreferencesModel::NAME() const { return Settings::value(SETTING).TO_FUNC; }
      |                                                              ^~~~~~~~~~~~~~~~
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:101:1: note: in expansion of macro ‘GET_PREF_FUNC_BOOL’
  101 | GET_PREF_FUNC_BOOL(     customThresholdScale,           Settings::USE_CUSTOM_THRESHOLD_SCALE                            )
      | ^~~~~~~~~~~~~~~~~~
In file included from /home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:1:
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.h:12:7: note: ‘class PreferencesModel’ defined here
   12 | class PreferencesModel : public PreferencesModelBase
      |       ^~~~~~~~~~~~~~~~
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:410:6: error: no declaration matches ‘void PreferencesModel::setCustomThresholdScale(bool)’
  410 | void PreferencesModel::setCustomThresholdScale(bool newCustomThresholdScale)
      |      ^~~~~~~~~~~~~~~~
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:410:6: note: no functions named ‘void PreferencesModel::setCustomThresholdScale(bool)’
In file included from /home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.cpp:1:
/home/runner/work/jasp-desktop/jasp-desktop/Desktop/gui/preferencesmodel.h:12:7: note: ‘class PreferencesModel’ defined here
   12 | class PreferencesModel : public PreferencesModelBase
      |       ^~~~~~~~~~~~~~~~
gmake[2]: *** [Desktop/CMakeFiles/JASP.dir/build.make:1326: Desktop/CMakeFiles/JASP.dir/gui/preferencesmodel.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:888: Desktop/CMakeFiles/JASP.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2
Error: Process completed with exit code 2.
```

**Comment by boutinb** [comment]
Forget again to update preferencesmodel.cpp.in in place of preferencesmodel.cpp...
It should work now.

File: Desktop/data/datasettablemodel.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ DataSetTableModel::DataSetTableModel(bool showInactive)
 {
 	
 	connect(DataSetPackage::pkg(),	&DataSetPackage::columnsFilteredCountChanged,	this, &DataSetTableModel::columnsFilteredCountChanged	);
-	connect(DataSetPackage::pkg(),	&DataSetPackage::columnDataTypeChanged,			this, &DataSetTableModel::columnTypeChanged				);
+	connect(DataSetPackage::pkg(),	&DataSetPackage::columnDataTypeChanged,			this, [&](QString colName) { emit columnTypeChanged(colName, int(DataSetPackage::pkg()->getColumnType(colName)));	});
```
Comment (JorisGoosen) [inline Desktop/data/datasettablemodel.cpp:28]: lekker

**Review by JorisGoosen — APPROVED** [review]
Much better like this

**Pull request merged**